src/main/java/gregtech/api/items/toolitem/IGTTool.java

```diff
@@ -304,10 +325,16 @@ default float getTotalAttackSpeed(ItemStack stack) {
 
     default int getTotalMaxDurability(ItemStack stack) {
         NBTTagCompound toolTag = getToolTag(stack);
+        IGTToolDefinition toolStats = getToolStats();
```

Comment:
If `stack` has `MAX_DURABILITY_KEY` you won't use this. So it would be better performance wise to call `getToolStats` only when you need it. Sure compiler may find this and optimize it for you, but it may not.